src/rabbit_variable_queue.erl

```diff
@@ -434,14 +434,20 @@
 
 
 %% we define the garbage collector threshold
-%% it needs to tune the GC calls inside `reduce_memory_use`
-%% see: rabbitmq-server-973 and `maybe_execute_gc` function
--define(DEFAULT_EXPLICIT_GC_RUN_OP_THRESHOLD, 250).
--define(EXPLICIT_GC_RUN_OP_THRESHOLD,
+%% it needs to tune the `reduce_memory_use` calls. Thus, the garbage collection.
+%% see: rabbitmq-server-973 and rabbitmq-server-964
+-define(DEFAULT_EXPLICIT_GC_RUN_OP_THRESHOLD, 1000).
+-define(EXPLICIT_GC_RUN_OP_THRESHOLD(Mode),
     case get(explicit_gc_run_operation_threshold) of
         undefined ->
-            Val = rabbit_misc:get_env(rabbit, lazy_queue_explicit_gc_run_operation_threshold,
-                ?DEFAULT_EXPLICIT_GC_RUN_OP_THRESHOLD),
+            Val = case Mode of
+                      lazy -> rabbit_misc:get_env(rabbit,
+                                                  lazy_queue_explicit_gc_run_operation_threshold,
+                                                  ?DEFAULT_EXPLICIT_GC_RUN_OP_THRESHOLD);
+                      _ -> rabbit_misc:get_env(rabbit,
+                                               queue_explicit_gc_run_operation_threshold,
+                                               ?DEFAULT_EXPLICIT_GC_RUN_OP_THRESHOLD)
+                  end,
```

Comment:
@dcorbacho minor thoughts here; how about a single function which takes desired env variable and calls `rabbit_misc:get_env/3`, to avoid what appears like code replication in both case clauses. Also, could be worth adding documentation for `queue_explicit_gc_run_operation_threshold` as part of this PR (unless `queue_explicit_gc_run_operation_threshold` configuration audience is not intended for general rabbit users).